provider/ec2/environ.go

```diff
@@ -411,6 +425,12 @@ func (e *environ) StartInstance(args environs.StartInstanceParams) (_ *environs.
 		if placement.availabilityZone.State != availableState {
 			return nil, errors.Errorf("availability zone %q is %q", placement.availabilityZone.Name, placement.availabilityZone.State)
 		}
+		if volumeAttachmentsZone != "" && volumeAttachmentsZone != placement.availabilityZone.Name {
+			return nil, errors.Errorf(
+				"placement %q conflicts with zone %q for volume attachment(s)",
```

Comment:
It's more that the ebs volumes have been created in a particular zone, right? And we can't attach an ebs volume create din zone A to a machine in zone B. Could the error be re-worded a bit to make it 100% clear to the user what the problem is? A user won't think in terms of volume attachments but ebs volumes. Could we also print the zones that the volumes are in?